### ISSUE TITLE
Join forms: No visual feedback when creating a new join form

### ISSUE THREAD

**Issue by ziggabyte** [zetkin/app.zetkin.org#2174]
## Description
When I create a new join form I get no visual feedback that something has been created, which has led me to click multiple times and (unknowingly) create many empty join forms. 

## Steps to reproduce
1. Go to `../people`
2. Click on the tab called "Join forms" 
3. Click on the "Create" button in the top right corner 
4. Select "Create new join form" 

### Expected Behaviour
I get some visual feedback that a new object has been created. 

### Actual Behaviour
The new join form is silently added to the bottom of the list

## Screenshots (if you have any)
If i click to create a new join form and my list of existing join forms looks like this, i will not see that the newly created one has been put at the bottom of the list, out of my view. 
![bild](https://github.com/user-attachments/assets/634d572a-cb2e-44ba-8a70-b805705af905)

## Suggestion
When we create other new activities or objects, we usually show the user the place where they can add information about the newly created thing. In this case, maybe opening the pane of the newly created join form so the user can start editing it?

**Comment by helihard** [comment]
Played around with this for a while, but abandoned because brain not braining properly after almost full day of code camp. 

However, designer team have two suggestions, hence I removed "needs-design" tag.

- Best would be that "Create new join form" here opens a dialog in the same way "Create person" and "Import people" does, and that a new form is created on "Create" in said dialog. That is, only once there has been some kind of editing such as changing the default title.
- Next best thing would be that "Create new join form" opens a pane in the same way clicking on the form in the forms list does. This could reportedly be achieved with useEffect in the JoinFormsPage.

**Comment by DaleLuce** [comment]
Looking into this

**Comment by richardolsson** [comment]
In #2206, @DaleLuce created a prototype using `useEffect()` that helped us understand the problem better. Here's (part of) [my comment](https://github.com/zetkin/app.zetkin.org/pull/2206#pullrequestreview-2335817855) on the prototype, that I think can be helpful to the next person to attempt this.

> My suggestion would be to try a completely different approach, similar to what we're doing for folders in the people list browser. For that, we use the store to keep track of the recently created folder. Try to search the codebase for recentlyCreatedFolder to see how that works.

**Comment by jones58** [comment]
Have looked into this. I think as well as highlighting the new join form, as we do when there is a new folder, we should also have a UI element "x join forms" next to "12 folders 32 lists" under the title People. This would mean we get an update even if we are looking at lists, or looking at folders etc.

**Comment by jones58** [comment]
Made a start on this, but had issues because of where joinforms have their store (not in the same place as views, folder and list). Sharing my code in case it's useful for others: https://github.com/jones58/app.zetkin.org/tree/joinformsissue

**Comment by katjam** [comment]
I'll take a look.

**Comment by katjam** [comment]
Because I ran into state management issues when trying to open the Join form edit `Pane` when arriving from another tab, I decided to split this issue so that we can improve the experience for when the Create form button is used on the Join tab.

After merge of after https://github.com/zetkin/app.zetkin.org/pull/2537... 

the behaviour for when I create a join form using the button in the people action menu is: 
- If I am not on the Join Forms tab, the Join form tab loads with the form added to the bottom of the list (same as old behaviour)
- If I am on the Join Forms tab, the Join form is added to the bottom of the list and a Join form edit pane opens so I can start editing the form right away.

**Comment by katjam** [comment]
@richardolsson I suggest that I open a new issue as something better worded but like:

On `organize/*/people` if I Create a Join Form from any tab, the Edit Pane for the new form opens. Current behavior is this only works if I am already on the Join Forms tab.
